### ISSUE TITLE
Notifications

### ISSUE THREAD

**Issue by Ahn1** [owncloud-archive/windows-phone#13]
We have to implement notifications for events.

It would be really cool

**Comment by framedynamic** [comment]
Did you mean live tile messages? AFAIK notifcations like incoming unread sms banners are only possible with a server application (which we are unable to run on via app).

First point should be made with PeriodicTask - but we have to be careful since such tasks are depleting battery very fast (at least on my 7.1 device - wp8 device seems not be affected so). Lightweight code needed.

**Comment by tcitworld** [comment]
I think toast notifications would be enough for a starting.

Using PeriodicTask needs maximum 5MB of memory. And the app must me lauched every 2 weeks, unless it doesn't stay.

**Comment by Ahn1** [comment]
mh, i thought there is a way to schedule a toast notification exactly. But i was wrong :/

Maybe we can get some oc core/calendar developer, which can do server side push notification. The api from microsoft is very simple. But i think this is nothing for the first Version.

When i go to "Administration" in my owncloud there is a section called "Cron". So i think there is already a implementation of periodic runned tasks in owncloud.
If we do this on that way whe had to transmit the Phone push url to the server. So we would need a new oc plugin       with a new api and a new db table...

I think for now, the periodic task would be the best.

**Comment by tcitworld** [comment]
Yes, but you can't force users have an internet access all the time. The only suitable way is to download events, and then do some sort of thing with them.

**Comment by Ahn1** [comment]
We already have the events of the user offline availible. So with a background task it would be a problem to raise notifications even offline.

The problem i see is that the ICal format does not have any limitations, when notifications can be.
WE can do them only every 15 minutes. With a push notification it would be much more exactly.

The other thing is, that whe dont know when the task runs the next time. So in the worst case the task run 10 minutes before the notification would raise. Then, 15 Minutes later the mobile is in battery saving mode. So the user dont get any notifications.
(Ok, while i wrote this i notice: Maybe this is not a critic point. Does push notifications work while battery saving mode? I dont think so).

Its very sad, that micosofts calendar do some magic things, so it is possible for them... :(

**Comment by framedynamic** [comment]
Batterymode will terminate push notifications too.

**Comment by tcitworld** [comment]
There may be information there about batterymode : http://www.neowin.net/forum/topic/1040493-battery-saver-does-it-stop-whatsapp-facebook-push-notifications/#entry594469687
